### PR TITLE
wrapping Errno::ETIMEDOUT in a Faraday::Error::TimeoutError exception (#322)

### PR DESCRIPTION
**Pull request by ricardovaleriano**
Hey guys, thank you for you AWESOME work.

I don't know if the tests are shaped in the best form, if you want this patch, but want it with a different test suite, just let me know and I will try to put it together.

This addresses the issue #322 in respect to wrapping the `Errno::ETIMEDOUT` in a `Faraday::Error::TimeoutError` exception.

### CODE REVIEW

**Comment by codesnik** [comment]
This is a really sad thread. No patch should be testes so rigorously that obvious very real problem isn't patched for a second year. If I'd have more time, I'd try to test it, but for now, I, as many others probably, going to write just

rescue Faraday::Error, Errno::ETIMEDOUT

File: test/adapters/integration.rb [hunk 1]
```diff
@@ -178,6 +178,17 @@ def test_timeout
         end
       end
 
+      def test_timeout_before_open
+        valid_but_inexistent_host = 'http://10.255.255.254'
```
Comment (mislav) [inline test/adapters/integration.rb]: We can't affort to maintain external dependencies in our test suite.

The proper way would be to spin up our own test server that's programmed to misbehave, i.e. delay accepting a connection. I'm not sure how to pull that off in Ruby.

**Comment by mislav** [comment]
@ricardovaleriano Thanks for your work on this. Sorry that it's so hard to make tests work consistently. It's one of the common problems in Faraday test suite because of so many adapters and Ruby versions that we need to support.

File: test/adapters/integration.rb (hunk 1 above)
Comment (kainosnoema) [inline test/adapters/integration.rb]: Why not just stub `http.request` to raise `Errno::ETIMEDOUT`?

**Comment by ricardovaleriano** [comment]
@mislav I totally understand your concerns man. And thank YOU for so much hard work on this. 
I'm not giving up yet, but I just don't know (yet) how to go further from here. =)

File: test/adapters/integration.rb (hunk 1 above)
Comment (mislav) [inline test/adapters/integration.rb]: I guess we are forced to revert to stubbing for now. @ricardovaleriano can you try that approach?

Comment (ricardovaleriano) [inline test/adapters/integration.rb]: > I guess we are forced to revert to stubbing for now. @ricardovaleriano can you try that approach?

Yep, give a couple of days and I'll take a look.

**Pull request closed without merging**